.gitignore

```diff
@@ -0,0 +1,7 @@
+# macOS environment
+.DS_Store
+# Xcode artifacts
+/.build
+/Packages
+/*.xcodeproj
```

Comment:
Should we ignore projects only within the repo's root dir?

Reply:
No, i can write `**/*.xcodeproj` but SPM does not automatically generate a *.xcodeproj file, so this rule can be removed, since we do not manually create this file.